pkg/oam/util/helper.go

```diff
@@ -869,6 +869,26 @@ func MergeMapOverrideWithDst(src, dst map[string]string) map[string]string {
 	return r
 }
 
+// ParseLabelString turns the label string into a map[string]string
+func ParseLabelString(labelString string) (map[string]string, error) {
+	if len(labelString) == 0 {
+		return nil, fmt.Errorf("the label string can't be empty")
+	}
+	labels := map[string]string{}
+	labelStrings := strings.Split(labelString, ",")
```

Comment:
Great, but if you are to write the label parser, remember to **remove spaces** inside the string first, as in `k8s.io/cli-runtime`:

https://github.com/kubernetes/cli-runtime/blob/4661c56a3b88e2288cba9b858718fca45f522e02/pkg/resource/builder.go#L480-L481

But I would suggest that you take a look at `k8s.io/apimahinery/pkg/labels` for some existing parsers, instead of writing your own:

https://github.com/kubernetes/apimachinery/blob/cff14a57b273bd82d64efc85aa0835aaf847a3ef/pkg/labels/labels.go#L146-L148